src/end_entity.rs

```diff
@@ -174,4 +176,17 @@ impl<'a> EndEntityCert<'a> {
             untrusted::Input::from(signature),
         )
     }
+
+    /// Returns a list of the DNS names provided in the subject alternative names extension
+    ///
+    /// This function must not be used to implement custom DNS name verification.
+    /// Verification functions are already provided as `verify_is_valid_for_dns_name`
+    /// and `verify_is_valid_for_at_least_one_dns_name`.
+    ///
+    /// Requires the `alloc` default feature; i.e. this isn't available in
+    /// `#![no_std]` configurations.
+    #[cfg(feature = "alloc")]
```

Comment:
My sense is that the `iterate_names()` API can/should be refactored to lean harder on iterators in a way that would be more idiomatic and can avoid allocations, but that's probably best left to a future followup PR.

Reply:
> My sense is that the iterate_names() API 

You mean `list_cert_dns_names` right? The `iterate_names` helper is already allocation free, it's the `list_cert_dns_names` function built on top that relies on a `Vec` and allocation. 

I filed https://github.com/rustls/webpki/issues/46 as a reminder (assuming I interpreted you correctly).